projects/views.py

```diff
@@ -1,3 +1,34 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from companies.permissions import IsCompanyMember
 
-# Create your views here.
+from projects.models import Project
+from projects.serializers import ProjectCreateUpdateSerializer, ProjectListSerializer
+
+class ProjectViewSet(ModelViewSet):
+    http_method_names = ('get', 'post', 'put', 'delete')
+    serializer_class = ProjectListSerializer
+    permission_classes = [IsAuthenticated]
+    queryset = Project.objects.all()
+    
+    def get_permissions(self):
+        match (self.action):
+            case 'update' | 'partial_update' | 'destroy':
+                return [IsAuthenticated(), IsCompanyMember()]
+            case _:       
+                return super().get_permissions()
+    
+    def get_queryset(self):
+        company_id = self.request.query_params.get("company_id")
```

Comment:
You're using query parameter `company_id`, but in your serializers and custom auth, you're injecting `company_id` into request from the JWT. It would be more secure and consistent to use:
```
company_id = getattr(self.request, "company_id", None)
```
This avoids leaking data if a user manually passes another company ID in the query.